src/osd/ReplicatedPG.cc

```diff
@@ -7075,7 +7075,17 @@ int ReplicatedPG::fill_in_copy_get(
       cursor.data_complete = true;
       dout(20) << " got data" << dendl;
     }
-    assert(cursor.data_offset <= oi.size);
```

Comment:
see https://github.com/ceph/ceph/pull/9203/files#diff-72747d40a424e7b5404366b557ff12a3L7053 . it could be greater than expected, if we are reading a bad replica which is larger than what object-info claims.